### ISSUE TITLE
vec_as_names not quiet when function is provided

### ISSUE THREAD

**Issue by smingerson** [r-lib/vctrs#1018]
According to docs in vctrs 0.2.4,
```
By default, the user is informed of any renaming caused by repairing the names.
 This only concerns unique and universal repairing. Set quiet to TRUE to silence the messages.
```
yet it also prints the message when providing  custom function.i

```r
vctrs::vec_as_names(c("a", "b"), repair = function(x) paste0(x, 
    "a"), quiet = FALSE)
#> New names:
#> * a -> aa
#> * b -> ba
#> [1] "aa" "ba"
```
Should the docs be changed to match the behavior, or the behavior to match the docs?

**Comment by lionel-** [comment]
This needs to be fixed, see also tidyverse/tibble#763
cc @krlmlr

**Comment by smingerson** [comment]
Thanks for pointing to the other conversation. I ran into the problem using `tidyr::pivot_wider()` and realized there is no convenient way to suppress the renaming functions. I will probably just rename after for now.
